File: src/main/kotlin/gbl/tag/type/GblMetadata.kt
package gbl.tag.type

import gbl.tag.GblType
import gbl.tag.TagHeader
import gbl.tag.Tag
import gbl.tag.TagWithHeader

data class GblMetadata(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    val metaData: ByteArray,
    override val tagData: ByteArray
): Tag, TagWithHeader {
    override fun copy(): Tag {
        return GblMetadata(
            tagHeader = tagHeader,
            tagType = tagType,
            metaData = metaData,
            tagData = arrayOf<Byte>().toByteArray()
        )
    }
}

File: python-library/tag/type/gbl_metadata.py
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ..gbl_type import GblType
    from ..tag_header import TagHeader
    from ..tag import Tag


@dataclass
class GblMetadata:
    """GBL Metadata tag"""

    tag_header: 'TagHeader'
    meta_data: bytes
    tag_data: bytes

    @property
    def tag_type(self) -> 'GblType':
        """Get tag type"""
        from ..gbl_type import GblType
        return GblType.METADATA

    def copy(self) -> 'Tag':
        """Create a copy of the tag"""
        return GblMetadata(
            tag_header=self.tag_header,
            meta_data=self.meta_data,
            tag_data=bytes()  # Empty byte array як у Kotlin
        )